llpc/lower/llpcSpirvLowerConstImmediateStore.cpp

```diff
@@ -223,16 +223,12 @@ bool SpirvLowerConstImmediateStore::tryProcessAlloca(AllocaInst *allocaInst) {
   }
 
   // Step 3: Create the global variable and replace the alloca
-  GlobalVariable *global = nullptr;
-  auto iter = m_allocToGlobals.find(initializer);
-  if (iter == m_allocToGlobals.end()) {
+  GlobalVariable *&global = m_allocToGlobals[initializer];
+  if (!global) {
     global = new GlobalVariable(*m_module, allocatedTy,
                                 true, // isConstant
                                 GlobalValue::InternalLinkage, initializer, "", nullptr, GlobalValue::NotThreadLocal,
                                 SPIRAS_Constant);
-    m_allocToGlobals[initializer] = global;
```

Comment:
Where is new global added to the map now ?

Reply:
You may run these lit tests (https://github.com/GPUOpen-Drivers/llpc/pull/2854) to verify the change.

Reply:
The new code is taking a *reference* to the global pointer stored in the map, so the assignment `global = new...` updates the map.